### ISSUE TITLE
PoolVector3Array::operator[] doesn't work for modification

### ISSUE THREAD

**Issue by garyo** [GodotNativeTools/godot-cpp#112]
The current definition is
Vector3 PoolVector3Array::operator [](const int idx)
```
{
	Vector3 v;
	*(godot_vector3 *) &v = godot::api->godot_pool_vector3_array_get(&_godot_array, idx);
	return v;
}
```
which returns a Vector3 by value. As far as I can tell, this violates the usual expectation of operator[] because you can't set the value through it. And it fails silently, which is fairly annoying.

I'd expect two variants:
```
          Vector3 & operator[](std::size_t idx)       { return something[idx]; }
    const Vector3 & operator[](std::size_t idx) const { return something[idx]; }
```
Looking at the godot::api functions though, I'm not even sure it's possible to implement this correctly at the moment. Because `godot_pool_vector3_array_get` also doesn't return a reference to the actual vector, it returns a copy. You have to use `...array_set` instead which is a bit unusual, right? 
Does the core Godot API even have a function to return a reference (non-const) to the actual Vector3 in question?

Maybe at least godot-cpp should only implement the const variant for now, so trying to do `vec3[idx] = newval;` will fail at compile time?

**Comment by karroffel** [comment]
PoolVectors need locking when writing, so returning a reference for writing is just asking for trouble. The `set` method can be used, which will lock and unlock for the write.

Alternatively, the `write` method can be used, which locks the vector but only unlocks when the Writer object leaves scope, so you can perform multiple writes.

**Comment by garyo** [comment]
Ah, I'll try that -- I noticed set() is really slow. (I have about 1M vertices to add.)
In that case, I'll submit a PR to make operator[] const. That way trying to write through it will be a compile-time error. (I already have that in my local version to ensure I don't try to do it.)

**Comment by garyo** [comment]
This is fixed by #116.
